server/utils/file/csvReader.js

```diff
@@ -9,51 +9,71 @@ export const createReaderFromStream = (stream, onHeaders = null, onRow = null, o
   const queue = new Queue()
 
   const start = () => new Promise((resolve, reject) => {
-    let started = false
     let ended = false
     let headers = null
     let total = 0
     let processingRow = false //prevents the call to processNext when a row is already being processed
 
+    /**
+     * Executes the specified function fn in a try catch.
+     * Calls "reject" if the execution throws an error.
+     */
+    const _tryOrCancel = async fn => {
+      try {
+        await fn()
+      } catch (e) {
+        cancel()
+        reject(e)
+      }
+    }
```

Comment:
maybe add this to functionsDefer, and rename it functionsUtils?

Reply:
good, I will do it